HPCC-16710 Document Dali and Sasha data store roles

g-pan opened the pull request:
Fix HPCC-16710 Document Dali and Sasha data store roles
Added content revolving around the Backup, and High Availability section
not all being on the same node as the Thor master.

Signed-off-by: G-Pan <greg.panagiotatos@lexisnexis.com>
@JamesDeFabia  FYI 
@RichardTaylorHPCC  please review
@jakesmith please review

g-pan committed:
HPCC-16710 Document Dali and Sasha data store roles

Fix HPCC-16710 Document Dali and Sasha data store roles
Added content revolving around the Backup, and High Availability section
not all being on the same node as the Thor master.

Signed-off-by: G-Pan <greg.panagiotatos@lexisnexis.com>

HPCCSmoketest commented:
Automated Smoketest
Sha: 3813f045edd5bc34990c42a7421af03c3bfcf802
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 91.605 seconds |

Install hpccsystems-platform-community_6.2.1-documentation0.el7.x86_64.rpm
HPCC Start: OK

Unittest result:

| total | passed | failed | timeout | 
|---|---|---|---|
|88 |88 |0 |0 |

HPCC Stop: OK
HPCC Uninstall: OK

jakesmith commented:
Sure.

1) > Sasha works independently of yet in conjunction with Dali

Should there be a comma between "...of, yet..." ?

2) Sasha.

As I previously mentioned (in mast last reply below), this section doesn't mention the coalescing aspect.
Which is probably the most vital role Sasha plays.

So it should say that near the top of the list of jobs it does in this section.

3) > Allocating a larger disk for Sasha reduces the amount of housekeeping needed

A little misleading..
Increasing the disk size alone doesn't help.

What helps the environment (Dali) is if Sasha's archiving configuration is tuned such that Dali doesn't keep too many workunits in memory.
i.e. increasing the amount that Sasha archives, for which it needs plenty of disk to do so.

NB: Not sure you should explicitly mention, but there's no limit on the total Sasha archive size, e.g. if it's been running for years, all those archived workunits will still exist on the Sasha node unless ops. has deleted them or moved them somewhere.

4) > An Explicit request to save using dalidiag:

capitalization, should say "An explicit .."

5) > All Dali SDS changes are recorded in a delta transaction log (in XML format) named convention of daliincXXXX.xml

Should say : "All Dali SDS changes are recorded in a delta transaction log (in XML format), with a naming convention of daliincXXXX.xml

6) > Once the workunits are archived they are no longer available in the Dali data store. The archives can still be retrieved

Perhaps you should say, they can still be accessed via Eclwatch, by restoring them to Dali.

7) Multiple Nodes

It should be emphasised that putting any of the components on nodes with slaves on is bad and leads to an unbalanced cluster, resulting in those slaves with less memory/cpu taking longer than the rest and dragging the whole performance of the cluster down as a result.

jakesmith commented:
@g-pan - a few comments

g-pan committed:
HPCC-16710 Document Dali and Sasha data store roles

Fix HPCC-16710 Document Dali and Sasha data store roles
Added content revolving around the Backup, and High Availability section
not all being on the same node as the Thor master.

Signed-off-by: G-Pan <greg.panagiotatos@lexisnexis.com>

HPCCSmoketest commented:
Automated Smoketest
Sha: 5007b0ec673f19567172adf986e8668cfa83a142
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 95.317 seconds |

Install hpccsystems-platform-community_6.2.1-documentation0.el7.x86_64.rpm
HPCC Start: OK

Unittest result:

| total | passed | failed | timeout | 
|---|---|---|---|
|88 |88 |0 |0 |

HPCC Stop: OK
HPCC Uninstall: OK

JamesDeFabia commented on docs/HPCCSystemAdmin/HPCCSystemAdministratorsGuide.xml:
```diff
@@ -175,34 +175,51 @@
+          files. It then presents options (via EclWatch) for their recovery or
+          deletion.</para>
+
+          <para>Sasha is the component responsible for removing expireds when
```
expireds should be expired files

JamesDeFabia commented on docs/HPCCSystemAdmin/HPCCSystemAdministratorsGuide.xml:
```diff
@@ -527,17 +529,85 @@
+      from Dali then processing it by archiving workunits (WUs) to disk. It is
+      beneficial to configure Sasha for a larger amount of archiving so that
+      Dali does not keep too many workunits in memory. This requires a larger
+      amount of diskspace. </para>
```
diskspace s/b disk space.

JamesDeFabia reviewed (changes requested):
Minor changes requirede

jakesmith commented:
@g-pan - changes look good.
But I see @JamesDeFabia has a couple of comments.

Back to you @g-pan - reassign when done.

g-pan committed:
HPCC-16710 Document Dali and Sasha data store roles

Fix HPCC-16710 Document Dali and Sasha data store roles
Added content revolving around the Backup, and High Availability section
not all being on the same node as the Thor master.

Signed-off-by: G-Pan <greg.panagiotatos@lexisnexis.com>

g-pan commented:
Requested changes made.  @jakesmith  please review.

jakesmith reviewed (approved):
Looks good.
@richardkchapman - ready to merge.

HPCCSmoketest commented:
Automated Smoketest
Sha: 68515f8158ea5f4133d1018cd8a794bdf334466f
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 91.754 seconds |

Install hpccsystems-platform-community_6.2.1-documentation0.el7.x86_64.rpm
HPCC Start: OK

Unittest result:

| total | passed | failed | timeout | 
|---|---|---|---|
|88 |88 |0 |0 |

HPCC Stop: OK
HPCC Uninstall: OK
